pnp/src/main/java/com/microsoft/azure/sdk/iot/pnphelpers/PnpHelper.java

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.sdk.iot.pnphelpers;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.microsoft.azure.sdk.iot.device.DeviceTwin.Property;
+import com.microsoft.azure.sdk.iot.device.Message;
+import lombok.NonNull;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.singleton;
+import static java.util.Collections.singletonMap;
+
+public class PnpHelper {
+
+    private static final String TELEMETRY_COMPONENT_NAME = "$.sub";
+    private static final String ENCODING_UTF_8 = StandardCharsets.UTF_8.name();
+    private static final String CONTENT_APPLICATION_JSON = "application/json";
+    private static final String PROPERTY_COMPONENT_IDENTIFIER_KEY = "__t";
+    private static final String PROPERTY_COMPONENT_IDENTIFIER_VALUE = "c";
+
+    private static final Gson gson = new Gson();
+    private static final GsonBuilder gsonBuilder = new GsonBuilder();
+
+    /**
+     * Create a plug and play compatible telemetry message.
+     * @param telemetryName The name of the telemetry, as defined in the DTDL interface. Must be 64 characters or less. For more details refer <see href="https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md#telemetry"
+     * @param telemetryValue The telemetry payload, in the format defined in the DTDL interface.
+     * @return A plug and play compatible telemetry message, which can be sent to IoT Hub.
+     */
+    public static Message createIotHubMessageUtf8(@NonNull String telemetryName, @NonNull Object telemetryValue) {
+        return createIotHubMessageUtf8(telemetryName, telemetryValue, null);
+    }
+
+    /**
+     * Create a plug and play compatible telemetry message.
+     * @param telemetryName The name of the telemetry, as defined in the DTDL interface. Must be 64 characters or less. For more details refer <see href="https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md#telemetry"
+     * @param telemetryValue The telemetry payload, in the format defined in the DTDL interface.
+     * @param componentName (optional) The name of the component in which the telemetry is defined. Can be null for telemetry defined under the root interface.
+     * @return A plug and play compatible telemetry message, which can be sent to IoT Hub.
+     */
+    public static Message createIotHubMessageUtf8(@NonNull String telemetryName, @NonNull Object telemetryValue, String componentName) {
+        Map<String, Object> payload = singletonMap(telemetryName, telemetryValue);
+        String messageId = java.util.UUID.randomUUID().toString();
```

Comment:
For what its worth, the SDK will assign a message id for you if you don't explicitly set one like this

Reply:
Ah, good to know! I'll get of this, then.